pan3d/explorers/contour.py

```diff
@@ -76,19 +75,48 @@ def __init__(self, server=None, local_rendering=None):
         if args.vtkjs:
             self.local_rendering = "vtkjs"
 
+        # Check if we have what we need
+        config_file = Path(args.import_state) if args.import_state else None
+        if (
+            (config_file is None or not config_file.exists())
+            and source is None
+            and xarray is None
+        ):
+            parser.print_help()
+            sys.exit(0)
+
         # setup
         self.last_field = None
         self.last_preset = None
-        self.xr = xr.open_dataset(fields_file, engine="netcdf4")
-        self.mesh = vtkDataSetReader(file_name=str(mesh_file))()
-        self._setup_vtk()
+        self._setup_vtk(xarray, source, config_file)
         self._build_ui()
 
     # -------------------------------------------------------------------------
     # VTK Setup
     # -------------------------------------------------------------------------
 
-    def _setup_vtk(self):
+    def _setup_vtk(self, xarray=None, source=None, import_state=None):
+        if xarray is not None:
+            self.source = vtkXArrayRectilinearSource(input=xarray)
+        elif source is not None:
+            self.source = source
+        elif import_state is not None:
+            self.source = vtkXArrayRectilinearSource()
+            config = json.loads(import_state.read_text())
+            self.source.load(config)
+            try:
+                field = config["preview"]["color_by"]
```

Comment:
```python
field = config.get("preview", {}).get("color_by")
```

Reply:
Should we have another name than "preview"?

Reply:
I don't see export state... It seems that the import/export is half backed.

Reply:
I think as it stands import/export state only happens through the main preview app.
None of our explorers export a state, but they are able to import a state. We could very-well add support to export state to them as well. I can work in it in the next milestone and it should not take too long.
Although, yes `preview` should be renamed to something more generic -- `rendering`?

Reply:
The globe explorer properly handle import/export.

Each explorer has its own namespace. "preview" is for the viewer, which is fine, unless we assume that piece of namespace get shared with all the rest of explorers.

Reply:
If we assume slice/contour can only import state, that could be fine too... We just need to know why we made that choice.

Reply:
The Globe Explorer, for the most part, was derived directly from the Preview app, so it got whatever came along with it. However, the export state functionality could be broken because I have not updated the stored essential variables in the state. The state export was omitted for the others to keep the UI and development simple and fast.
A standard UI component for import-export can make things easier by adding callbacks from the explorers.